services/kanto/src/kanto_service.cpp

```diff
@@ -26,7 +28,77 @@ void KantoService::OnStop()
 
 int32_t KantoService::StartServices()
 {
-    KANTO_HILOGI(KANTO_MODULE_SERVICE, "starting kanto!");
+    KANTO_HILOGI(KANTO_MODULE_SERVICE, "Starting Kanto Service");
+
+    std::string ip = "127.0.0.1:1883";
+    std::string id = "temperature_sensor";
+    mqtt::client client(ip, id);
+
+    try {
+        KANTO_HILOGD(KANTO_MODULE_SERVICE, "Connecting to MQTT broker");
+        client.connect();
+
+        KANTO_HILOGD(KANTO_MODULE_SERVICE, "Subscribing to edge/thing/response");
+        client.subscribe("edge/thing/response");
+
+        KANTO_HILOGD(KANTO_MODULE_SERVICE, "Requesting device data");
+        mqtt::message_ptr deviceDataRequestMessagePointer = mqtt::make_message("edge/thing/request", "");
+        client.publish(deviceDataRequestMessagePointer);
+
+        auto deviceDataResponseMessagePointer = client.consume_message();
+
+        if (!deviceDataResponseMessagePointer) 
+        {
+            KANTO_HILOGI(KANTO_MODULE_SERVICE, "No device data available");
+            return 1;
+        }
+
+        KANTO_HILOGD(KANTO_MODULE_SERVICE, "Unsubscribing from edge/thing/response");
+        client.unsubscribe("edge/thing/response");
+        
+        KANTO_HILOGI(KANTO_MODULE_SERVICE, "Device data received %s", deviceDataResponseMessagePointer->get_payload_str().c_str());
+
+        nlohmann::json deviceData = nlohmann::json::parse(deviceDataResponseMessagePointer->get_payload_str());
+        std::string tenantId = deviceData["tenantId"];
+        std::string gatewayDeviceId = deviceData["deviceId"];
+        std::string airConditionerDeviceId = gatewayDeviceId + ":AC";
+        std::string deviceIdInNamespaceNotation = airConditionerDeviceId;
+        int namespaceSeparatorIndex = deviceIdInNamespaceNotation.find(":");
+        deviceIdInNamespaceNotation.replace(namespaceSeparatorIndex, 1, "/");
+        std::string telemetryTopic = "t/" + tenantId + "/" + airConditionerDeviceId;
+
+        nlohmann::json telemetryMessagePayload = {
```

Comment:
With this hint I assume we are talking about hits file: https://github.com/nlohmann/json/blob/develop/single_include/nlohmann/json.hpp ?

We should note in the license description MIT next to Apache 2.0 now.

Reply:
that is correct. Added MIT license to LICENSES now